### PR TITLE
Free food script

### PR DESCRIPTION
**Pull request by michaelpeng**
Additionally added a standup intro response which, in retrospect, is painfully bad. I also now have Party in the USA stuck in my head. Great.

### CODE REVIEW

File: scripts/freefood.coffee [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Description:
+#   Announce free food.
+#
+# Commands:
+#   <botname> free <food> <location>
+
+module.exports = (robot) ->
+  robot.respond /free (.+)/i, (msg) ->
```
Comment (NevilleS) [inline scripts/freefood.coffee]: You can tighten up this regex to match the food & location into separate match groups, then you don't need to mangle the string as much

File: scripts/standup.coffee [hunk 2]
```diff
@@ -90,6 +90,7 @@ module.exports = (robot) ->
       "Andup-stay Ime-tay!",
       (() -> "Initiate STANDUP protocol (#{(Math.random() * 100000).toString(16)})... BEGIN"),
       (() -> ([1..(Math.floor(Math.random() * 10) + 1)].map (i) -> "STAND!").join(" ")),
+      "So we do our STAND UP, let's all come along, the butterflies fly awayyy!!",
```
Comment (NevilleS) [inline scripts/standup.coffee]: Wow

File: scripts/freefood.coffee [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          "There's free #{food} upstairs! Go, go, go!",
+          "Free #{food} upstairs! PUSH AND SHOVE, PEOPLE!",
+          "Quick! Get your #{food} upstairs!",
+          "Free #{food} upstairs! Use the stairs, why doncha.",
+          "BZZT. Instance of free [<#{food.toUpperCase()}>] detected. Location: [<UPSTAIRS>].",
+          "And I go back upstairs all the time. Turns out freedom ain't nothing but missing free food. Wishing I'd realized what I had when you were mine...",
+          "If you had...one shot...one opportunity...to seize all the free #{food} you ever wanted upstairs...would you capture it, or just let it slip?",
+        ]
+      else
+        console.log("Invalid free food location!")
```
Comment (NevilleS) [inline scripts/freefood.coffee]: Probably better to have the bot say something in chat about not understanding you :smile:

File: scripts/freefood.coffee (hunk 1 above)
Comment (michaelpeng) [inline scripts/freefood.coffee]: Yeah I initially was trying to use regex to get match groups/captures but couldn't find clear documentation so I gave up and just messed around with the string :sob: I'll give it another go.

Comment (NevilleS) [inline scripts/freefood.coffee]: What you have works quite well, you'll still need the magic removal of in/the/at stuff, but match groups are just a good thing to know :+1:

Comment (michaelpeng) [inline scripts/freefood.coffee]: Yeah agreed, I almost have it working and that's what I ended up deciding as well :P

**Comment by michaelpeng** [comment]
Ready for further review :dancers:

File: scripts/freefood.coffee [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    food = food.join(" ").trim()
+
+    switch location
+      when "kitchen"
+        news = [
+          "Free #{food} in the kitchen!",
+          "Free #{food} in the kitchen!",
+          "To the kitchen for free #{food}!",
+          "To the kitchen for free #{food}!",
+          "THEY'RE COOKING UP SOME FREE #{food.toUpperCase()} in the kitchen! GO GO GO!"
```
Comment (NevilleS) [inline scripts/freefood.coffee]: "in the kitchen".toUpperCase()

Comment (michaelpeng) [inline scripts/freefood.coffee]: Caps is love, caps is life.

**Comment by NevilleS** [comment]
Also check this:

```
Hubot> hubot free in kitchen
Hubot> Responding to message: 'hubot free in kitchen'
Free  in the kitchen!
```

**Comment by NevilleS** [comment]
:+1: :shipit: ?

**Comment by michaelpeng** [comment]
:+1: :shipit: !

**Pull request merged**